components/genome-browser-view.js

```diff
@@ -0,0 +1,146 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import PropTypes from "prop-types";
+import {
+  DataArea,
+  DataAreaTitle,
+  DataItemLabel,
+  DataItemValue,
+  DataPanel,
+} from "./data-area";
+import GenomeBrowser from "./genome-browser";
+import {
+  GenomeBrowserFacets,
+  filterByAllSelectedFilters,
+} from "./genome-browser-facets";
+import { ChevronLeftIcon, ChevronRightIcon } from "@heroicons/react/20/solid";
+
+// number of files to display on genome browser
+const NUMBER_OF_FILES_FOR_DISPLAY = 20;
+
+/**
+ * This is the view for display data genome browser for a variant.
+ * It contains facets for filter the files, genome browser to display the files and pagination if needed.
+ */
+export function GenomeBrowserView({ files, assembly, coordinates }) {
+  const router = useRouter();
+  useEffect(() => {
+    const isBrowser = router.asPath.endsWith(`#!browser`);
+    setShowBrowserData(isBrowser);
+  }, [router]);
+  const [showBrowserData, setShowBrowserData] = useState(false);
+  const [filteredFiles, setFilteredFiles] = useState(files);
+  const [selectedFilters, setSelectedFilters] = useState([]);
+  const [totalPage, setTotalPage] = useState(
+    Math.ceil(filteredFiles.length / NUMBER_OF_FILES_FOR_DISPLAY)
+  );
+  const [currentPage, setCurrentPage] = useState(1);
+  const [displayedFiles, setDisplayedFiles] = useState(
+    filteredFiles.slice(0, 20)
+  );
```

Comment:
This is probably fine, but if you have time-consuming `useState` initializers, you can use something like `useState(() => filteredFiles.slice(0, 20))` so that `filteredFiles` only gets called when the owning component (`GenomeBrowserView` in this case) gets mounted in the DOM, but not after. As it stands, `filteredFiles.slice(0, 20)` gets called every rerender, even though its result then gets thrown away after the first render. Again, that’s probably fine in this case, as it probably doesn’t take much time.

Reply:
Got it. Thanks!